### PR TITLE
PT-1891 Fixed mongodb-summary connection with ssl

### PR DESCRIPTION
**Pull request by percona-csalguero**
- Added SSL connection options
- Fixed old tests
- Replaced gofmt by gofumpt in Makefile
- There are no ssl test for mongodb-summary because the current sandbox doesnt support it

### CODE REVIEW

**Review by askomorokhov — APPROVED** [review]

File: src/go/mongolib/proto/replconfig.go [hunk 1]
```diff
@@ -4,8 +4,10 @@ import (
 	"go.mongodb.org/mongo-driver/bson/primitive"
 )
 
-type ReplicaSetConfigTags map[string]string
-type GetLastErrorModes map[string]*ReplicaSetConfigTags
+type (
```
Comment (BupycHuk) [inline src/go/mongolib/proto/replconfig.go:7]: I would suggest to split them by blank line since they are not related

File: src/go/pt-mongodb-summary/main.go [hunk 2]
```diff
@@ -39,18 +44,23 @@ const (
 	DefaultOutputFormat       = "text"
 	typeMongos                = "mongos"
 
-	// Exit Codes
+	// Exit Codes.
 	cannotFormatResults              = 1
 	cannotParseCommandLineParameters = 2
 	cannotGetHostInfo                = 3
-	cannotGetReplicasetMembers       = 4
+	cannotGetClientOptions           = 4
```
Comment (BupycHuk) [inline src/go/pt-mongodb-summary/main.go:51]: Are you sure that it's a good idea to change exit codes in minor changes?

File: src/go/pt-mongodb-summary/main.go [hunk 3]
```diff
@@ -201,34 +218,52 @@ func main() {
...
+
+		return
+	}
+
 	client, err := mongo.NewClient(clientOptions)
 	if err != nil {
-		log.Fatalf("Cannot get a MongoDB client: %s", err)
+		exitStatus = cannotConnectToMongoDB
+
+		log.Errorf("Cannot get a MongoDB client: %s", err)
```
Comment (BupycHuk) [inline src/go/pt-mongodb-summary/main.go:222]: why don't you exit after this?

File: src/go/pt-mongodb-summary/main.go [hunk 4]
```diff
@@ -201,34 +218,52 @@ func main() {
...
 	}
+
 	if err := client.Connect(ctx); err != nil {
-		log.Fatalf("Cannot connect to MongoDB: %s", err)
+		exitStatus = cannotConnectToMongoDB
+
+		log.Errorf("Cannot connect to MongoDB: %s", err)
```
Comment (BupycHuk) [inline src/go/pt-mongodb-summary/main.go:228]: and this

File: src/go/pt-mongodb-summary/main.go [hunk 5]
```diff
@@ -161,17 +172,22 @@ type collectedInfo struct {
 }
 
 func main() {
+	var exitStatus int
+
+	defer func() {
+		os.Exit(exitStatus)
```
Comment (BupycHuk) [inline src/go/pt-mongodb-summary/main.go]: previous usage looks simpler, let's use `os.Exit()` instead of `exitStatus` and `defer`

File: src/go/pt-mongodb-summary/main.go [hunk 6]
```diff
@@ -920,27 +998,88 @@ func getChunksCount(ctx context.Context, client *mongo.Client) ([]proto.ChunksBy
...
+			return nil, err
+		}
+		roots.AppendCertsFromPEM(ca)
+		tlsConfig.RootCAs = roots
+	}
+
+	return tlsConfig, nil
+}
+
+func expandHome(path string) string {
```
Comment (BupycHuk) [inline src/go/pt-mongodb-summary/main.go:1061]: can we rename it to absolutePath?

**Review by BupycHuk — CHANGES_REQUESTED** [review]
Mostly looks good to me, but have a few questions. Please answer to them.

File: src/go/pt-mongodb-summary/main.go (hunk 2 above)
Comment (percona-csalguero) [reply src/go/pt-mongodb-summary/main.go:51]: It wasn't being used so there is no way that anyone had that error code.

File: src/go/pt-mongodb-summary/main.go (hunk 6 above)
Comment (percona-csalguero) [reply src/go/pt-mongodb-summary/main.go:1061]: We could but:
1) It might not be absolute.
2) Even Go libraries use `path` as the variable name: https://golang.org/pkg/path/filepath/#Clean

**Review by BupycHuk — APPROVED** [review]
